Code/Framework/AzToolsFramework/AzToolsFramework/AssetBrowser/AssetBrowserTableModel.cpp

```diff
@@ -24,13 +24,23 @@ namespace AzToolsFramework
             AZ_Assert(
                 m_filterModel,
                 "Error in AssetBrowserTableModel initialization, class expects source model to be an AssetBrowserFilterModel.");
+
             QSortFilterProxyModel::setSourceModel(sourceModel);
 
-            connect(m_filterModel, &QAbstractItemModel::rowsInserted, this, &AssetBrowserTableModel::UpdateTableModelMaps);
-            connect(m_filterModel, &QAbstractItemModel::rowsRemoved, this, &AssetBrowserTableModel::UpdateTableModelMaps);
-            connect(m_filterModel, &QAbstractItemModel::layoutChanged, this, &AssetBrowserTableModel::UpdateTableModelMaps);
+            auto checkUpdateModelTimer = [&]()
+            {
+                if (m_updateModelMapTimerId > 0)
+                {
+                    killTimer(m_updateModelMapTimerId);
+                }
+                m_updateModelMapTimerId = startTimer(250);
```

Comment:
nit: `250` could be stored in a constant

```
const int ModelRefreshWaitTime = 250;

auto checkUpdateModelTimer = [&]()
    {
        if (m_updateModelMapTimerId > 0)
        {
            killTimer(m_updateModelMapTimerId);
        }
        m_updateModelMapTimerId = startTimer(ModelRefreshWaitTime);
    }
);
```

Reply:
Nice. I'll get a revision up with that.